benchmarks/shared/src/main/scala/higherkindness/mu/rpc/benchmarks/AvroBenchmark.scala

```diff
@@ -18,52 +18,52 @@ package higherkindness.mu.rpc.benchmarks
 
 import java.util.concurrent.TimeUnit
 
-import cats.effect.{IO, Resource}
-import higherkindness.mu.rpc.benchmarks.shared.Runtime
+import cats.effect.IO
+import higherkindness.mu.rpc.ChannelForAddress
 import higherkindness.mu.rpc.benchmarks.shared.Utils._
 import higherkindness.mu.rpc.benchmarks.shared.models._
 import higherkindness.mu.rpc.benchmarks.shared.protocols.PersonServiceAvro
-import higherkindness.mu.rpc.benchmarks.shared.server._
+import higherkindness.mu.rpc.benchmarks.shared._
 import higherkindness.mu.rpc.protocol.Empty
-import higherkindness.mu.rpc.testing.servers.ServerChannel
 import org.openjdk.jmh.annotations._
 
 @State(Scope.Thread)
 @BenchmarkMode(Array(Mode.Throughput))
 @OutputTimeUnit(TimeUnit.SECONDS)
-class AvroBenchmark extends Runtime {
+class AvroBenchmark extends ServerRuntime {
 
-  implicit val handler: AvroHandler[IO] = new AvroHandler[IO]
-
-  def clientIO: Resource[IO, PersonServiceAvro[IO]] =
-    Resource
-      .liftF(PersonServiceAvro.bindService[IO])
-      .flatMap(ServerChannel[IO](_))
-      .flatMap(sc => PersonServiceAvro.clientFromChannel[IO](IO(sc.channel)))
+  @Setup
+  def setup(): Unit = startServer
 
   @TearDown
-  def shutdown(): Unit = {}
+  def shutdown(): Unit = stopServer
+
+  def clientCall[B](f: PersonServiceAvro[IO] => IO[B]): B =
+    PersonServiceAvro
+      .client[IO](ChannelForAddress("localhost", grpcPort))
+      .use(f)
+      .unsafeRunTimed(defaultTimeOut)
+      .get
 
   @Benchmark
   def listPersons: PersonList =
-    clientIO.use(_.listPersons(Empty)).unsafeRunTimed(defaultTimeOut).get
+    clientCall(_.listPersons(Empty))
```

Comment:
This builds (and tears down) the client every time it's called, which involves creating object pools and various other expensive things. I didn't look too closely but there's a lot going on in [AbstractManagedChannelImplBuilder](https://github.com/grpc/grpc-java/blob/459cbc7b44e8e14748dbfbd4c47a2958f34778e9/core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java), so it's probably pretty slow.

Maybe a more realistic benchmark would be to create the client once during the setup phase and re-use it. That matches what most users would do in production: create the `Resource` and call `use` on it once at the edge of their app, and pass the client around wherever it's needed.

Shall we take a look at doing this in a separate PR?

Reply:
Thanks Chris, you are completely right.

I've tried to fix it in https://github.com/higherkindness/mu-scala/pull/884/commits/7957b3ff85e01b2d981610d9dc71c44162a94a67, what do you think about doing it that way?

Reply:
LGTM 👍 I'm interested to see the benchmark results after this change.

Reply:
They seem to be a bit faster now :)